lighthouse-core/report/html/renderer/performance-category-renderer.js

```diff
@@ -298,6 +309,68 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
     element.appendChild(passedElem);
     return element;
   }
+
+  /**
+   * Render the control to filter the audits by metric. The filtering is done at runtime by CSS only
+   * @param {LH.ReportResult.AuditRef[]} filterableMetrics
+   * @param {HTMLDivElement} element
+   */
+  renderMetricAuditFilter(filterableMetrics, element) {
+    filterableMetrics.unshift(/** @type {LH.ReportResult.AuditRef} */ ({acronym: 'All'}));
+    // thx https://codepen.io/surjithctly/pen/weEJvX
+    const filterBarEl = this.dom.createElement('div', 'lh-filterbar');
+    const filterBarSummaryEl = this.dom.createChildOf(filterBarEl, 'span', 'lh-filterbar__summary');
+    filterBarSummaryEl.textContent = 'Filter to relevant audits: ';
+    const labelSelectors = [];
+    const auditSelectors = [];
+    for (const metric of filterableMetrics) {
+      // appended into `element` for sweet ~ selectors to work
+      const elemId = `metric-${metric.acronym}`;
+      const radioEl = this.dom.createChildOf(element, 'input', 'lh-filterbar__radio', {
+        type: 'radio',
+        name: 'metricsfilter',
+        id: elemId,
+        hidden: 'true',
+      });
+      const labelEl = this.dom.createChildOf(filterBarEl, 'label', 'lh-filterbar__label', {
+        for: elemId,
+      });
+      labelEl.textContent = metric.acronym || metric.id;
+      if (metric.acronym === 'All') {
+        radioEl.checked = true;
+      }
+      // Dynamically write some CSS, for the CSS-only filtering
+      labelSelectors.push(`.lh-filterbar__radio#${elemId}:checked ~ .lh-filterbar > .lh-filterbar__label[for="${elemId}"]`); // eslint-disable-line max-len
+      if (metric.relevantAudits) {
+        /* Generate some CSS selectors like this:
+            #metric-CLS:checked ~ .lh-audit-group > #layout-shift-elements,
+            #metric-CLS:checked ~ .lh-audit-group > #non-composited-animations,
+            #metric-CLS:checked ~ .lh-audit-group > #unsized-images
+        */
+        auditSelectors.push(metric.relevantAudits.map(auditId => `#${elemId}:checked ~ .lh-audit-group > #${auditId}`).join(',\n')); // eslint-disable-line max-len
+      }
+    }
+
+    const styleEl = this.dom.createChildOf(filterBarEl, 'style');
+    styleEl.textContent = `
+      ${labelSelectors.join(',\n')} {
+        background: var(--color-blue-A700);
+        color: var(--color-white);
+      }
+      /* If selecting non-All, hide all audits (and also the group header/description… */
+      .lh-filterbar__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit,
+      .lh-filterbar__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit-group__description,
+      .lh-filterbar__radio:checked:not(#metric-All) ~ .lh-audit-group .lh-audit-group__itemcount {
+        display: none;
```

Comment:
Are there alternatives to _hiding_ the filtered audits? I think keeping things in a constant place is beneficial. wdyt of reduce opacity?